onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -601,6 +605,104 @@ TEST(CApiTest, io_binding) {
   binding.ClearBoundOutputs();
 }
 
+#ifdef USE_CUDA
+TEST(CApiTest, io_binding_cuda) {
+  Ort::SessionOptions session_options;
+  Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CUDA(session_options, 0));
+  Ort::Session session(*ort_env, MODEL_URI, session_options);
+
+  Ort::MemoryInfo info_cuda("Cuda", OrtAllocatorType::OrtArenaAllocator, 0, OrtMemTypeDefault);
+  Ort::Allocator cuda_allocator(session, info_cuda);
+  auto allocator_info = cuda_allocator.GetInfo();
+  ASSERT_TRUE(info_cuda == allocator_info);
+
+  const std::array<int64_t, 2> x_shape = {3, 2};
+  std::array<float, 3 * 2> x_values = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f};
+  void* input_data = cuda_allocator.Alloc(x_values.size() * sizeof(float));
+  ASSERT_NE(input_data, nullptr);
+  cudaMemcpy(input_data, x_values.data(), sizeof(float) * x_values.size(), cudaMemcpyHostToDevice);
```

Comment:
I expected to have to make some cmake changes to get the shared library project link with the cuda runtime library given that the definition for cudaMemcpy(...) lives there, but this change as is worked for me locally without any cmake changes. If anyone has some insight, please share...